### PR TITLE
feat(store/mem_file): add in-memory eds file implementation

### PR DESCRIPTION
**Pull request by walldiss**
Adds in-memory file implementation

### CODE REVIEW

File: share/getter.go [hunk 1]
```diff
@@ -73,15 +73,15 @@ func (ns NamespacedShares) Verify(root *Root, namespace Namespace) error {
 
 	for i, row := range ns {
 		// verify row data against row hash from original root
-		if !row.verify(originalRoots[i], namespace) {
+		if !row.Verify(originalRoots[i], namespace) {
```
Comment (Wondertan) [inline share/getter.go:76]: A minor change from local changes. It makes sure we don't segfault when proof is malformed.
```suggestion
		if row.Proof == nil && row.Shares == nil {
			return fmt.Errorf("row verification failed: no proofs and shares")
		}
		if !row.Verify(originalRoots[i], namespace) {
```

File: share/store/mem_file.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	err = batchAdder.Commit()
+	if err != nil {
+		return share.NamespacedRow{}, err
+	}
+
+	cid := ipld.MustCidFromNamespacedSha256(root)
+	row, proof, err := ipld.GetSharesByNamespace(context.TODO(), bserv, cid, namespace, len(shares))
+	if err != nil {
+		return share.NamespacedRow{}, err
+	}
```
Comment (Wondertan) [inline share/store/mem_file.go:81]: We might want to dedup this later

File: share/store/mem_file.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+func (f *MemFile) axis(axisType rsmt2d.Axis, axisIdx int) []share.Share {
+	switch axisType {
+	case rsmt2d.Row:
+		return f.Eds.Row(uint(axisIdx))
+	case rsmt2d.Col:
+		return f.Eds.Col(uint(axisIdx))
+	default:
+		panic("unknown axis")
+	}
+}
```
Comment (Wondertan) [inline share/store/mem_file.go:101]: I want this to be an rsmt2d method so bad

Comment (Wondertan) [reply share/store/mem_file.go:101]: (no actionable here)

File: share/store/mem_file_test.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds/edstest"
+	"github.com/celestiaorg/celestia-node/share/sharetest"
+)
+
+func TestMemFileShare(t *testing.T) {
+	eds := edstest.RandEDS(t, 32)
```
Comment (Wondertan) [inline share/store/mem_file_test.go:22]: kind big for the test but doesn't really matter

File: share/store/mem_file_test.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	if to < from {
+		from, to = to, from
+	}
+	expected := shares[from]
+	namespace := share.GetNamespace(expected)
+
+	// change namespace for all shares in range
+	for i := from; i <= to; i++ {
+		shares[i] = expected
+	}
```
Comment (Wondertan) [inline share/store/mem_file_test.go:73]: I would leave a todo to make this a helper. I have tests where I need something similar

File: share/store/mem_file.go (hunk 2 above)
Comment (walldiss) [reply share/store/mem_file.go:81]: Later

File: share/store/mem_file_test.go (hunk 5 above)
Comment (walldiss) [reply share/store/mem_file_test.go]: I've extracted it to test pkg in last commit for you 🙂

**Pull request merged**